src/locales/de/parsers/DETimeUnitRelativeFormatParser.ts

```diff
@@ -5,35 +5,47 @@ import { AbstractParserWithWordBoundaryChecking } from "../../../common/parsers/
 import { reverseTimeUnits } from "../../../utils/timeunits";
 import { matchAnyPattern } from "../../../utils/pattern";
 
-export default class DETimeUnitAgoFormatParser extends AbstractParserWithWordBoundaryChecking {
+export default class DETimeUnitRelativeFormatParser extends AbstractParserWithWordBoundaryChecking {
     constructor() {
         super();
     }
 
     innerPattern(): RegExp {
         return new RegExp(
-            `(?:\\s*((?:nächste|kommende|folgende|letzte|vergangene|vorige|vor(?:her|an)gegangene)(?:s|n|m|r)?|vor|in)\\s*)?` +
+            `(?:((?:(?:über)(?:über)nächste|kommende|folgende|(?:vor)?(?:vor)?letzte|vergangene|vorige|vor(?:her|an)gegangene)(?:s|n|m|r)?|vor|in|nach|ab\\sjetzt(?:\\sin)?)\\s*)?` +
                 `(${NUMBER_PATTERN})?` +
                 `(?:\\s*(nächste|kommende|folgende|letzte|vergangene|vorige|vor(?:her|an)gegangene)(?:s|n|m|r)?)?` +
-                `\\s*(${matchAnyPattern(TIME_UNIT_DICTIONARY)})`,
+                `\\s*(${matchAnyPattern(TIME_UNIT_DICTIONARY)}\\b)` +
+                `(?:\\s*(früher|frueher|vorher|davor|zuvor|später|spaeter|danach|nachher|ab\\sjetzt))?`,
             "i"
         );
     }
 
     innerExtract(context: ParsingContext, match: RegExpMatchArray) {
-        const num = match[2] ? parseNumberPattern(match[2]) : 1;
-        const unit = TIME_UNIT_DICTIONARY[match[4].toLowerCase()];
-        let timeUnits = {};
-        timeUnits[unit] = num;
-
         // Modifier
-        let modifier = match[1] || match[3] || "";
+        let modifier = match[1] || match[3] || match[5] || "";
         modifier = modifier.toLowerCase();
         if (!modifier) {
             return;
         }
 
-        if (/vor/.test(modifier) || /letzte/.test(modifier) || /vergangen/.test(modifier)) {
+        let num = match[2] ? parseNumberPattern(match[2]) : 1;
+        if (/vorvorletzte/.test(modifier) || /überübernächste/.test(modifier)) {
+            num += 2;
+        } else if (/vorletzte/.test(modifier) || /übernächste/.test(modifier)) {
+            num += 1;
+        }
+
+        const unit = TIME_UNIT_DICTIONARY[match[4].toLowerCase()];
+        let timeUnits = {};
```

Comment:
Nit: first, I'd rename `num` to `value` and I think `let timeUnits = { unit: num };` is clearer to understand.